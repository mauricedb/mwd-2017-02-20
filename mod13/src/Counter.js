import React, { Component } from 'react';
import Hello from './Hello';

class CounterPresentation extends Component {
    constructor(){
        super();
        this.onClick = this.onClick.bind(this);
    }
    
    onClick(){
        this.props.increment();
        this.props.increment();
        this.props.increment();
    }

    render() {
        console.log('Rendering Counter')
        return (
            <div>
                <Hello name={"Maurice"} />
                <div>
                    <button onClick={this.onClick}>Add</button>
                </div>
                <div>
                    Count: {this.props.counter}
                </div>
                
            </div>
        );
    }
}

class CounterCountainer extends Component {
    constructor(){
        super();
        this.state = {
            counter: 0
        }

        this.increment = this.increment.bind(this)
    }

    increment(){
        // this.setState({counter: this.state.counter + 1});
        this.setState(state => ({counter: state.counter + 1}));
    }


    render() {
        return <CounterPresentation counter={this.state.counter} increment={this.increment} />;
        }
}



export default CounterCountainer;