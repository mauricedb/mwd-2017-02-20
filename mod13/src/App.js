import React, { Component } from 'react';
import './App.css';
import Counter from './Counter';
import Cats from './Cats';
import Header from './Header';

class App extends Component {
  render() {
    return (
      <div className="App">
        <Header name="Maurice" />
        <p className="App-intro">
          To get started, edit <code>src/App.js</code> and save to reload.
        </p>
        <Counter />
        <Cats />
      </div>
    );
  }
}

export default App;
