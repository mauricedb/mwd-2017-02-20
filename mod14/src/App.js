import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';
import Page1 from './Page1';
import Page2 from './Page2';

class App extends Component {
  constructor(){
    super();
    this.state = {
      page: 0
    }
    this.gotoPage = this.gotoPage.bind(this);  
  }

  gotoPage(page){
    this.setState({page});
  }

  render() {
    return (
      <div className="App">
        <div className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h2>Welcome to React</h2>
        </div>
        <p className="App-intro">
          To get started, edit <code>src/App.js</code> and save to reload.
        </p>
        <a href="#" onClick={() => this.gotoPage(1)}>Page 1</a>
        &nbsp;
        <a href="#" onClick={() => this.gotoPage(2)}>Page 2</a>

        { this.state.page === 1 ? <Page1 />: null}
        { this.state.page === 2 ? <Page2 />: null}
      </div>
    );
  }
}

export default App;
