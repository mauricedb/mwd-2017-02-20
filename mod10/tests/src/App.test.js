import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import App from './App';
import renderer from 'react-test-renderer';
import Counter from './Counter';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
});


it('renders an `.App-logo`', () => {
    const wrapper = shallow(<App />);
    expect(wrapper.find('img.App-logo').length).toBe(1);
  });


  it ('Can incement', () =>{

    const wrapper = shallow(<Counter />)
    wrapper.find('button').simulate('click');
    wrapper.find('button').simulate('click');

   expect(wrapper.state('counter')).toBe(2);

  })


it('App renders correctly', () => {
  const tree = renderer.create(
    <App now="12:15" />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});