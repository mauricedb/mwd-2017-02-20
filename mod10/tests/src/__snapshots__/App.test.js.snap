exports[`test App renders correctly 1`] = `
<div
  className="App">
  12:15
  <div
    className="App-header">
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg" />
    <h2>
      Welcome to React
    </h2>
  </div>
  <p
    className="app-intro">
    To get started, edit 
    <code>
      src/App.js
    </code>
     and save to reload.
  </p>
</div>
`;
